packages/next/src/build/webpack/plugins/nextjs-require-cache-hot-reloader.ts

```diff
@@ -96,35 +96,40 @@ export class NextJsRequireCacheHotReloader implements WebpackPluginInstance {
       clearModuleContext(targetPath)
     })
 
-    compiler.hooks.afterEmit.tap(PLUGIN_NAME, (compilation) => {
-      RUNTIME_NAMES.forEach((name) => {
-        const runtimeChunkPath = path.join(
-          compilation.outputOptions.path!,
-          `${name}.js`
-        )
-        deleteCache(runtimeChunkPath)
-      })
-      let hasAppPath = false
-
-      // we need to make sure to clear all server entries from cache
-      // since they can have a stale webpack-runtime cache
-      // which needs to always be in-sync
-      const entries = [...compilation.entries.keys()].filter((entry) => {
-        const isAppPath = entry.toString().startsWith('app/')
-        hasAppPath = hasAppPath || isAppPath
-        return entry.toString().startsWith('pages/') || isAppPath
-      })
-
-      if (hasAppPath) {
+    compiler.hooks.afterEmit.tapAsync(
+      PLUGIN_NAME,
+      async (compilation, callback) => {
+        try {
+          for (const name of RUNTIME_NAMES) {
+            const runtimeChunkPath = path.join(
+              compilation.outputOptions.path!,
+              `${name}.js`
+            )
+            await deleteCache(runtimeChunkPath)
```

Comment:
Should this be run in parallel/with a semaphore? Seems it waits for each individual call.

Also: can we process the full list of paths in a single IPC call instead of many?

Reply:
Updated in https://github.com/vercel/next.js/pull/52492/commits/b9d78217ab8b93d8c84bb60e8235f368f2c1f6fe